packages/shared-ux/chrome/navigation/src/ui/components/navigation_section_ui.tsx

```diff
@@ -221,12 +243,18 @@ const nodeToEuiCollapsibleNavProps = (
     return { items, isVisible: true };
   }
 
-  const onClick = (e: React.MouseEvent) => {
+  const onClick = (e: React.MouseEvent<HTMLElement | HTMLButtonElement>) => {
+    if (customOnClick) {
+      customOnClick(e);
```

Comment:
do we need to prevent default and/or stop propagation or is the idea that it has to be handled by the CB? 

I see that below propagation is stopped even if customOnClick is passed

Reply:
We don't want to prevent default for the customOnClick as the consumer handles it.
Regarding the stopPropagation calls, those were probably added to fix some behaviour but it works fine without them. I removed those in https://github.com/elastic/kibana/pull/183157/commits/891581cfb4a7abd151c2964af0063e7da3f5b1fe

Thanks for pointing that out 👍